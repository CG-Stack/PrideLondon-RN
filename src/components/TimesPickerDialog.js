// @flow
import React from "react";
import CheckBox from "./CheckBox";
import Dialog from "./Dialog";

type Props = {
  onTimesSelected: Function,
  onCancel: Function,
  visible: boolean
};

type State = {
  morning: boolean,
  afternoon: boolean,
  evening: boolean
};

class TimesPickerDialog extends React.PureComponent<Props, State> {
  state = {
    morning: true,
    afternoon: true,
    evening: true
  };

  toggleMorning = () => {
    this.setState({ morning: !this.state.morning });
  };

  toggleAfternoon = () => {
    this.setState({ afternoon: !this.state.afternoon });
  };

  toggleEvening = () => {
    this.setState({ evening: !this.state.evening });
  };

  apply = () => {
    const times = [];
    if (this.state.morning) {
      times.push("Morning");
    }

    if (this.state.afternoon) {
      times.push("Afternoon");
    }

    if (this.state.evening) {
      times.push("Evening");
    }

    if (times.length > 0) {
      this.props.onTimesSelected(times);
    }
  };

  render() {
    return (
      <Dialog
        applyButtonText="Show XX events"
        onApply={this.apply}
        onCancel={this.props.onCancel}
        title="Select time"
        visible={this.props.visible}
      >
        <CheckBox
          onChange={this.toggleMorning}
          checked={this.state.morning}
          label="Morning"
        />
        <CheckBox
          onChange={this.toggleAfternoon}
          checked={this.state.afternoon}
          label="Afternoon"
        />
        <CheckBox
          onChange={this.toggleEvening}
          checked={this.state.evening}
          label="Evening"
        />
      </Dialog>
    );
  }
}

export default TimesPickerDialog;
