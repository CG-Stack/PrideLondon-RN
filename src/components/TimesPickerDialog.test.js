// @flow
import React from "react";
import { shallow } from "enzyme";
import TimesPickerDialog from "./TimesPickerDialog";

it("renders correctly", () => {
  const output = shallow(
    <TimesPickerDialog onTimesSelected={() => {}} onCancel={() => {}} visible />
  );
  expect(output).toMatchSnapshot();
});
