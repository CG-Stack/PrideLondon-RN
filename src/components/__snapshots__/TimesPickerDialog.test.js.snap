// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Dialog
  applyButtonText="Show XX events"
  onApply={[Function]}
  onCancel={[Function]}
  title="Select time"
  visible={true}
>
  <CheckBox
    checked={true}
    label="Morning"
    onChange={[Function]}
  />
  <CheckBox
    checked={true}
    label="Afternoon"
    onChange={[Function]}
  />
  <CheckBox
    checked={true}
    label="Evening"
    onChange={[Function]}
  />
</Dialog>
`;
